File: 2024/rust/day-02/src/bin/part2.rs
use std::{fs::File, io};

fn main() -> Result<(), Box<dyn std::error::Error>> {
    match solve("inputs/input2.txt") {
        Ok(res) => println!("{}", res),
        Err(e) => panic!("{}", e)
    }

    Ok(())
}

fn solve(input_file_path: &str) -> Result<i64, Box<dyn std::error::Error>> {
    let file = File::open(input_file_path)?;
    let file_reader = io::BufReader::new(file);

    Ok(todo!())
}

#[cfg(test)]
mod tests {
    use crate::solve;

    #[test]
    fn part2() {
        let expected_res: i64 = 4;

        match solve("inputs/test2.txt") {
            Ok(res) => assert_eq!(expected_res, res),
            Err(e) => panic!("{}", e)
        }
    }
}

File: 2025/go/day-01/part2.go
package main

import (
	"bufio"
	"fmt"
	"os"
)

func main() {
	result, err := solve("inputs/input2.txt")
	if err != nil {
		panic(err)
	}
	fmt.Println(result)
}

func solve(inputPath string) (int, error) {
	file, err := os.Open(inputPath)
	if err != nil {
		return 0, err
	}
	defer file.Close()

	scanner := bufio.NewScanner(file)
	
	// TODO: Implement solution
	for scanner.Scan() {
		line := scanner.Text()
		_ = line
	}

	if err := scanner.Err(); err != nil {
		return 0, err
	}

	return 0, nil
}
